agent/engine/docker_task_engine.go

```diff
@@ -1018,6 +1018,14 @@ func (engine *DockerTaskEngine) createContainer(task *apitask.Task, container *a
 		}
 	}
 
+	if container.ShouldCreateWithEnvFiles() {
+		err := task.ReadEnvVarsFromEnvfiles(container)
```

Comment:
I feel it would be better for the name to be **MergeEnvVarsFromEnvFiles**, it makes me confusing for a second that where you merge the env vars until I read the function logic.

Reply:
We don't have the precedence requirements for secrets and the env file, right?

Reply:
i think the requirement is just that all existing env vars (in this case secret as well) have precendence - though it's highly unlikely that someone will create a secret but then also store it as plaintext in s3..or is it 🤔

Reply:
also renamed the method.